api/v1/verticadb_types.go

```diff
@@ -801,6 +801,50 @@ type Subcluster struct {
 	// State to indicate whether the operator must shut down the subcluster
 	// and not try to restart it.
 	Shutdown bool `json:"shutdown,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// Create client proxy pods for the subcluster if defined
+	// All the subcluster connections will be redirected to the proxy pods
+	Proxy Proxy `json:"proxy,omitempty"`
+}
+
+type Proxy struct {
+	// +kubebuilder:validation:required
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// The docker image name that contains the Vertica proxy server.
+	Image string `json:"image"`
+
+	// +kubebuilder:default:=1
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// The number of replicas that the proxy server will have.
+	Replica int32 `json:"replica,omitempty"`
```

Comment:
Good suggestion. Set the replica as (sc.size % 2) currently so the ratio would be 1:2.

Reply:
More like 1:1(1 proxy for each vertica pod), 1:2(1 proxy for 2 vertica pods). I really like this idea but let's stick with replica for now. We can revisit this later. I don't have enough info about the recommended resources for the proxy. As it is a bottleneck we need to implement an autoscaler which, unlike the VerticaAutoscaler, will always be running. @fenic-fawkes needs to be part of the discussion, it can even be an objective for next PI.